packages/shared/src/common/normalize-zip.ts

```diff
@@ -4,8 +4,17 @@
  * @returns The normalized zip code as a string.
  */
 export function normalizeZipCode(zipCode: string): string {
-  if (zipCode.includes("-") && zipCode.length === 9) {
-    return zipCode.slice(0, 4);
+  if (!isValidZipCode(zipCode)) {
+    throw new Error("Zip codes may only contain numbers (0-9) and a dash (-)");
   }
+  if (zipCode.includes("-") && zipCode.length === 9) return zipCode.slice(0, 4);
+  if (zipCode.length === 8) return zipCode.slice(0, 4);
   return zipCode.slice(0, 5);
 }
+
+function isValidZipCode(zipCode: string): boolean {
+  if (zipCode.length === 0) return false;
+  const regex = /[^0-9-]/;
+  const res = regex.test(zipCode);
+  return !res;
```

Comment:
Not a huge deal, but usually its easier to understand code that doesn't do boolean inversion.

In this case, we're returning `true` for "is it valid" if the response is false for the test that it contains something that's not what's valid.

We could rewrite it so it returns `true` for "is it valid" if the response is true for the test that contains only valid contents.

Regex: `^[0-9-]+$`

Reply:
Yea, this is a good point. I was struggling even when writing the function haha that was a good indicator that a change is required